### PR TITLE
FEAT: test services

### PR DESCRIPTION
**Pull request by naftalima**
## Objective and Motivation
-  The form that associates the pokemons to the battle.

### LogicTeamPokemonTest
| Test     | Description  | Assert     |
| :---        |    :----   |   :--- |
| <code>test_position_unique </code>  |  It has no repeated position. | True   |
| <code>test_position_repeated </code>   | has repeated position. | False  |
| <code>test_pokemons_unique </code>  | It has no repeated pokemon.  | True   |
| <code>test_pokemons_repeated </code>   | It has repeated pokemon. | False  |
| <code>test_team_sum_valid </code>  | The sum of the pokemon team points is less than 600. | True   |
| <code>test_team_sum_invalid </code>   | The sum of points of the pokemon team is more than 600. | False  |

### LogicBattleTest
| Test     | Description  | Assert     |
| :---        |    :----   |  :--- |
| <code>test_get_winner </code>  | Happy Path  | True   |
| <code>test_opponent_wins_in_a_tie </code>   | In case of a tie, the challenged user wins. | <ul><li>True </li><li> <code>assertEqual</code>, The winner is the opponent  </li></ul> |

### CODE REVIEW

File: backend/battles/tests/test_services.py [hunk 1]
```diff
@@ -281,3 +289,263 @@ def test_send_battle_result(self, email_mock):
...
+            fake_json = 1
+            if pokemon_name == "venusaur":
+                fake_json = {
+                    "defense": 165,
+                    "attack": 145,
+                    "hp": 235,
+                    "name": "venusaur",
+                    "img_url": "https://raw.githubusercontent.com"
+                    "/PokeAPI/sprites/master/sprites/pokemon/25.png",
+                    "pokemon_id": 179,
```
Comment (rsarai) [inline backend/battles/tests/test_services.py:370]: you changed everything to be `poke_id`, why is it different here?

File: backend/battles/tests/test_services.py [hunk 2]
```diff
@@ -281,3 +289,263 @@ def test_send_battle_result(self, email_mock):
...
+
+        self.add_pokemons_to_team(
+            pokemons=pokemons[0:2], positions=[3, 2, 1], team=self.team_creator
+        )
+        self.add_pokemons_to_team(
+            pokemons=pokemons[1:3], positions=[1, 2, 3], team=self.team_opponent
+        )
+        winner = get_winner(self.battle)
+        self.assertTrue(winner)
+        self.assertEqual(winner, self.opponent)
```
Comment (rsarai) [inline backend/battles/tests/test_services.py:540]: The opponent still is the winner, something about this setup is still wrong

**Review by rsarai — CHANGES_REQUESTED** [review]

File: backend/battles/tests/test_services.py (hunk 1 above)
Comment (naftalima) [reply backend/battles/tests/test_services.py:370]: I forgot to change.

File: backend/battles/tests/test_services.py (hunk 2 above)
Comment (naftalima) [reply backend/battles/tests/test_services.py]: I fixed it. I was putting the wrong intervals to slice the list

**Review by rsarai — APPROVED** [review]

**Pull request merged**